### ISSUE TITLE
Singularity: do not include .singularity.d in the Docker container

### ISSUE THREAD

**Issue by dstndstn** [gwastro/pycbc#3184]
Hi,

I was trying to run your docker container in singularity, but it fails:
```
$ singularity build pycbc.sif docker://pycbc/pycbc-el7:latest
...
ERROR:   build: failed to make environment symlinks: symlink .singularity.d/runscript /gpfs/dlang/singularity-tmp/rootfs-94270e0b-6480-11ea-b821-000af7d3f980/singularity: file exists
FATAL:   While performing build: packer failed to pack: while inserting base environment: build: failed to make environment symlinks: symlink .singularity.d/runscript /gpfs/dlang/singularity-tmp/rootfs-94270e0b-6480-11ea-b821-000af7d3f980/singularity: file exists
```

Turns out, this is due to including your own `.singularity.d` in the container;
```
https://github.com/gwastro/pycbc/blob/7f383fa61eb5457f2d52ea93ab58856497821ccf/Dockerfile#L3
```

as diagnosed by the Singularity folks, in this ticket:
https://github.com/sylabs/singularity/issues/5113

**Comment by ahnitz** [comment]
Thanks for posting this @dstndstn, I've assigned someone more familiar with our docker setup than me.

**Comment by josh-willis** [comment]
This is done because of how singularity images are built and distributed to CVMFS for the open science grid, as done here:
[cvmfs-singularity-sync](https://github.com/opensciencegrid/cvmfs-singularity-sync/blob/master/cvmfs-singularity-sync)
Though it is possible that at some point this may move to a direct invocation of singularity, until it does, I don't think we can avoid doing this.

**Comment by dstndstn** [comment]
I don't know anything about that, but the Singularity folks say this is not recommended (see  https://github.com/sylabs/singularity/issues/5113#issuecomment-598300858 )

"This isn't something that should be done in a Dockerfile."

**Comment by dstndstn** [comment]
and it makes your docker container un-runnable in current singularity (3.5.3)

**Comment by josh-willis** [comment]
This should now be resolved with the `pycbc-el7:latest` Dockerhub image, and with any future releases *after* the current release, v1.16.9.  There is no way to retroactively change this on earlier releases.

I am closing this ticket, if you encounter this or similar problems on a release after v1.16.9, please open another ticket.

Closed via #3379
